pkg/fanal/analyzer/language/python/packaging/packaging.go

```diff
@@ -34,35 +50,122 @@ var (
+			}
 
-	// .egg file is zip format and PKG-INFO needs to be extracted from the zip file.
-	if strings.HasSuffix(input.FilePath, ".egg") {
-		pkginfoInZip, err := a.analyzeEggZip(input.Content, input.Info.Size())
+			r = pkginfoInZip
+		}
+		app, err := a.parse(path, r)
 		if err != nil {
-			return nil, xerrors.Errorf("egg analysis error: %w", err)
+			return xerrors.Errorf("parse error: %w", err)
+		} else if app == nil {
+			return nil
 		}
 
-		// Egg archive may not contain required files, then we will get nil. Skip this archives
-		if pkginfoInZip == nil {
-			return nil, nil
+		apps = append(apps, *app)
+		return nil
+	})
+
+	for i := range apps {
+		if err := a.fillAdditionalData(input.FS, &apps[i]); err != nil {
+			log.Logger.Warnf("Unable to collect additional info: %s", err)
 		}
+	}
 
-		r = pkginfoInZip
+	if err != nil {
+		return nil, xerrors.Errorf("python package walk error: %w", err)
 	}
+	return &analyzer.AnalysisResult{
+		Applications: apps,
+	}, nil
+}
 
-	p := packaging.NewParser()
-	return language.AnalyzePackage(types.PythonPkg, input.FilePath, r, p, input.Options.FileChecksum)
+func (a packagingAnalyzer) fillAdditionalData(fsys fs.FS, app *types.Application) error {
+
+	if len(app.Libraries) > 0 {
+		var licenses []string
+		for _, lic := range app.Libraries[0].Licenses {
+			// Parser adds `file://` prefix to filepath from `License-File` field
+			// We need to read this file to find licenses
+			// Otherwise, this is the name of the license
+			if !strings.HasPrefix(lic, "file://") {
+				licenses = append(licenses, lic)
+				continue
+			}
+			licenseFielPath := filepath.Base(strings.TrimPrefix(lic, "file://"))
+
+			findings, err := classifyLicense(app.FilePath, licenseFielPath, a.licenseClassifierConfidenceLevel, fsys)
+			if err != nil {
+				return err
+			}
+			// License found
+			if len(findings) > 0 {
+				foundLicenses := lo.Map(findings, func(finding types.LicenseFinding, _ int) string {
+					return finding.Name
+				})
+				licenses = append(licenses, foundLicenses...)
+			}
+		}
+
+		app.Libraries[0].Licenses = licenses
+	}
+	return nil
+}
+
+func classifyLicense(dir string, licPath string, classifierConfidenceLevel float64, fsys fs.FS) (types.LicenseFindings, error) {
+	// Note that fs.FS is always slashed regardless of the platform,
+	// and path.Join should be used rather than filepath.Join.
+	f, err := fsys.Open(path.Join(filepath.Dir(dir), licPath))
+	if errors.Is(err, fs.ErrNotExist) {
+		return nil, nil
+	} else if err != nil {
+		return nil, xerrors.Errorf("file open error: %w", err)
+	}
+	defer f.Close()
+
+	l, err := licensing.Classify(licPath, f, classifierConfidenceLevel)
+	if err != nil {
+		return nil, xerrors.Errorf("license classify error: %w", err)
+	}
+
+	if l == nil {
+		return nil, nil
+	}
+
+	return l.Findings, nil
+}
+
+func (a packagingAnalyzer) parse(path string, r dio.ReadSeekerAt) (*types.Application, error) {
+	return language.Parse(types.PythonPkg, path, r, a.pkgParser)
```

Comment:
we need to use logic for packages: 

```suggestion
	return language.ParsePackage(types.PythonPkg, path, r, a.pkgParser, fileChecksum)
```

Don't forget take checksum flag from `PostAnalysisInput.AnalysisOptions.FileChecksum`

Reply:
fixed